### PR TITLE
Validate the configuration produced by LlamaTune inverse_transform()

### PR DESCRIPTION
**Pull request by bpkroth**
Revamp of #806 with the fix as well.

Adds code to handle appropriate scaling and type conversion and checking of the inverted config.

Splits that code out to a separate method for individual configs to make pylint happy and improve code readability.

### CODE REVIEW

**Comment by bpkroth** [comment]
@kkanellis please have a look at this when you get a chance.  Pretty sure the clipping is the appropriate thing to do, but another set of eyes would be good.  Thanks!

File: mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 # pylint: disable=redefined-outer-name
 
+# TODO: Add quantized tunables here for better test coverage.
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py:18]: To be handled in a future PR

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py:18]: #813

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py]: Actually, that change was not that bad so I just added it here as well.

File: mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py [hunk 2]
```diff
@@ -60,7 +60,8 @@
                 "type": "int",
                 "default": 2000000,
                 "range": [0, 1000000000],
-                "log": false
+                "log": false,
+                "quantization": 100
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py:64]: Makes sure we test quantized values with our default tunable groups fixtures.
The change was actually quite small, so I included it here as well, but I can split it back out again instead too.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py]: Moved that over to #814

File: mlos_core/mlos_core/spaces/adapters/llamatune.py [hunk 3]
```diff
@@ -261,7 +352,7 @@ def _transform(self, configuration: dict) -> dict:
             # Clip value to force it to fall in [0, 1]
             # NOTE: HeSBO projection ensures that theoretically but due to
             #       floating point ops nuances this is not always guaranteed
-            value = max(0.0, min(1.0, norm_value))  # pylint: disable=redefined-loop-name
+            value = max(0.0, min(1.0, norm_value))
```
Comment (motus) [inline mlos_core/mlos_core/spaces/adapters/llamatune.py:355]:
```suggestion
            value = np.clip(norm_value, 0, 1)
```

File: mlos_core/mlos_core/spaces/adapters/llamatune.py [hunk 4]
```diff
@@ -5,8 +5,10 @@
 """Implementation of LlamaTune space adapter."""
 from typing import Dict, Optional
 from warnings import warn
+import os
```
Comment (motus) [inline mlos_core/mlos_core/spaces/adapters/llamatune.py:8]:
```suggestion
import os
from typing import Dict, Optional
from warnings import warn
```
to make `isort` happy

File: mlos_core/mlos_core/spaces/adapters/llamatune.py [hunk 5]
```diff
@@ -123,28 +125,111 @@ def inverse_transform(self, configurations: pd.DataFrame) -> pd.DataFrame:
...
+        configuration_dict = dict(config)
+        double_checked_config = self._transform(dict(target_config))
+        double_checked_config = {
+            # Skip the special values that aren't in the original space.
+            k: v
+            for k, v in double_checked_config.items()
+            if k in configuration_dict
+        }
+        if double_checked_config != configuration_dict and (
+            os.environ.get("MLOS_DEBUG") in {"1", "true", "y", "yes"}
```
Comment (motus) [inline mlos_core/mlos_core/spaces/adapters/llamatune.py:211]:
```suggestion
            os.environ.get("MLOS_DEBUG", "false").lower() in {"1", "true", "y", "yes"}
```

**Review by motus — APPROVED** [review]
pushed a few small fixes; will merge into master and of @kkanellis has some updates, we'll do it in the subsequent PRs.

**Pull request merged**